client/state/billing-transactions/types.ts

```diff
@@ -14,6 +14,7 @@ export interface BillingTransaction {
 	address: string;
 	amount: string;
 	tax_country_code: string;
+	tax_external_id: string;
```

Comment:
Because the server is not quite yet providing `tax_external_id` (and because it can be null currently; I've left a comment on the patch), it might help if we set this type to be optional so that TypeScript can protect us in case it's not set. You're already correctly guarding for it with the code `transaction.tax_external_id && transaction.tax_external_id.length > 0` but this way TS will also help us out.

```suggestion
	tax_external_id?: string;
```

Reply:
Thank you, great catch! 🙌🏻